orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.groovy

```diff
@@ -639,7 +639,7 @@ class JedisExecutionRepository implements ExecutionRepository {
         stage.parentStageId = map["stage.${stageId}.parentStageId".toString()]
         stage.requisiteStageRefIds = map["stage.${stageId}.requisiteStageRefIds".toString()]?.tokenize(",") ?: emptySet()
         stage.scheduledTime = map["stage.${stageId}.scheduledTime".toString()]?.toLong()
-        stage.context = map["stage.${stageId}.context".toString()] ? mapper.readValue(map["stage.${stageId}.context".toString()], MAP_STRING_TO_OBJECT) : emptyMap()
+        stage.context = map["stage.${stageId}.context".toString()] ? new StageContext(stage, mapper.readValue(map["stage.${stageId}.context".toString()], MAP_STRING_TO_OBJECT)) : new StageContext(stage)
```

Comment:
Repository needs to construct a `StageContext` so that ancestor chaining works on retrieved stages.